string ball_holder 
ModelAtGoalRadius model_at_goal_radius